addons/purchase/models/account_invoice.py

```diff
@@ -36,7 +36,7 @@ def _prepare_invoice_line_from_po_line(self, line):
         else:
             qty = line.qty_received - line.qty_invoiced
         if float_compare(qty, 0.0, precision_rounding=line.product_uom.rounding) <= 0:
-            qty = 0.0
+            qty = qty
```

Comment:
I mean, without looking at anything else than the code. 

if WhatEverConditionIWant : 
 qty = qty

Just makes no sense, just remove both lines, you'll have the same behavior ...